static/js/apps/nl_interface/query_history.tsx

```diff
@@ -22,9 +22,14 @@ import axios from "axios";
 import _ from "lodash";
 import React, { useEffect, useState } from "react";
 
-const MAX_QUERY_COUNT = 20;
+const MAX_QUERY_COUNT = 10;
 
-export function QueryHistory(): JSX.Element {
+interface QueryHistoryProps {
+  // Callback function for history item clicks.
+  onItemClick: (query: string) => void;
```

Comment:
nit: Wonder if we could make the link display the first query (as it is now), but the URL include full query chain?  eg, instead of `result.push(queryList[0]);` below, add the whole list

Reply:
ah yes, that would be good.

Reply:
will do that in a follow up